### ISSUE TITLE
Using pretrained XLM for underresourced lanuages?

### ISSUE THREAD

**Issue by kochbj** [facebookresearch/XLM#128]
Hi all,

I am trying to help a friend find a solution to translate from English to Amharic (and eventually other related semitic languages).  There is currently only about 200,000 English-Amharic sentences in the Opus database. I don't have a lot of NLP experience. Is this a feasible task with XLM or is the training data too small? If feasible, would you recommend trying train the MLM and TLM models from scratch on this data (perhaps enriching with a similar language)  or finetuning using a pretrained model? If you do recommend enriching with Arabic or Hebrew, can you provide any more guidlines about how to go about doing this?

Thanks for your advice; I really appreciate it!

Best,
B

**Comment by sebastian-nehrdich** [comment]
Well I have nothing to do with the development of XLM, but I can report from my own experience that it is possible to train a decent XLM model of you have enough monolingual data of each language available. So in our scenario, we had a small corpus of Sanskrit (100mb raw text) and a somewhat larger corpus of Tibetan (400mb of raw text) and about 100.000 parallel sentences (half as much as you have).
Going from Sanskrit to Tibetan we achieved 30 BLEU which is the first and thus also the highest ever achieved. I suspect though that in our scenario, because we have a lot of monolingual data each but only very little bilingual data, we overfit on the bilingual data very very early. Basicly we were training masked prediction monolingual and bilingual at the same time from the beginning. It's not the best way to do it I assume because the real power of monolingual pretraining does not come into play. So I advise you to try for example to train first only monolingual masked prediction for a while and then switch to bilingual - maybe that will work. We discussed this problem here in another issue before: https://github.com/facebookresearch/XLM/issues/100
I always wanted to try to do some monolingual training and then later switch to bilingual training, but I never found the time to try it yet.
